addons/web/static/src/js/views/form/form_renderer.js

```diff
@@ -234,7 +234,7 @@ var FormRenderer = BasicRenderer.extend({
      * @param {Object} state the result from a getLocalState call
      */
     setLocalState: function (state) {
-        for (const notebook of this.el.querySelectorAll(':scope div.o_notebook')) {
+        for (const notebook of this.el.querySelectorAll(':scope div.o_notebook:not(div.o_notebook div.o_notebook)')) {
```

Comment:
Looks like this checks that notebooks aren't nested, but for all we know you could have a `div.o_notebook` in an html field that is not inside a notebook, and this would still match it (when it shouldn't). A better fix would probably be to check that the notebook is not inside a field widget:
```suggestion
        for (const notebook of this.el.querySelectorAll(':scope div.o_notebook')) {
            if (notebook.closest(".o_field_widget") {
                continue;
            }
        }
```

Also, this is missing a test :)